ir/v1.cpp

```diff
@@ -131,14 +133,10 @@ unsigned IR::Primitive::inferOperandTypes() const {
 const IR::Type *IR::Primitive::inferOperandType(int operand) const {
     if (name == "truncate")
         return IR::Type::Bits::get(32);
-    if ((name == "count" || name == "execute_meter") && operand == 1) {
-        if (auto obj = operands[0]->to<IR::GlobalRef>()) {
-            if (auto tbl = obj->obj->to<IR::Stateful>()) {
-                if (tbl->instance_count > 0) {
-                    int width = ceil_log2(tbl->instance_count);
-                    return IR::Type::Bits::get(width); } } } }
-    if (name.startsWith("execute_stateful") && operand == 1) {
-            return IR::Type::Bits::get(32); }
+    if ((name == "count" || name == "execute_meter") && operand == 1)
+        return IR::Type::Bits::get(32);
```

Comment:
Interestingly, if you examine the test failures, we were always casting to bit<32> anyway when we actually pass the operands to the extern method. For example, before we had:

```p4
     @name(".m_action") action m_action_0(bit<14> idx) {
         my_indirect_counter.count((bit<32>)idx);
     }
```

After this patch, we have:

```p4
     @name(".m_action") action m_action_0(bit<32> idx) {
         my_indirect_counter.count(idx);
     }
```

The second version is consistent with what `p4c-bm` generates.